Code/Framework/AzToolsFramework/AzToolsFramework/Entity/EditorEntityContextBus.h

```diff
@@ -162,9 +162,9 @@ namespace AzToolsFramework
         virtual ~EditorEntityContextNotification() = default;
 
         /// Called before the context is reset.
-        virtual void PrepareForContextReset() {}
+        virtual void OnPrepareForContextReset() {}
```

Comment:
A deprecation notice might be needed if we are renaming this since there can be other usages outside the o3de repo. Maybe force 'PrefapareForContextReset()' to call 'OnPrepareForContextReset()?' and add a deprecation notice?

Personally, I think 'PrepareForContextReset()' is a good enough name, but if the reasoning behind the renaming is to start notification events with 'On...' , I understand.

Reply:
That's likely not a bad call, I was on the fence about if it was worth it or not as this existing API I don't think was even really used by our code, so it did seem very unlikely it would be used by any external code, but it's likely better to be safe than sorry so I'll create one to track.

I found renaming to `OnPrepareForContextReset` was worth it as it helped disambiguate the separate virtual call in `EntityContext` called the same thing (which isn't easily visible in the PR).

Reply:
Created https://github.com/o3de/o3de/issues/15310